compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1138,7 +1139,14 @@ private void defineRecordTypeNode(BLangRecordTypeNode recordTypeNode, SymbolEnv
             return;
         }
 
+        List<BType> fieldTypes = new ArrayList<>();
+        for (BField field : recordType.fields.values()) {
+            BType type = field.type;
+            fieldTypes.add(type);
+        }
+
```

Comment:
we can call `symResolver.markParameterizedType(recordType, recordType.fields.values());` always and call again for
`symResolver.markParameterizedType(recordType, recordType.restFieldType);` that way we don't need a new array list.

Reply:
```java
recordType.fields.values()
``` 
returns a collection of `BField` objects. What I need are the types of those `BField` objects.